sdks/swift/Mentat/Mentat/Core/TypedValue.swift

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Mentatlib
+
+class TypedValue: OptionalRustObject {
+
+    var valueType: ValueType {
+        return typed_value_value_type(self.raw!)
+    }
+
+    func asLong() -> Int64 {
+        defer {
+            self.raw = nil
+        }
+        return typed_value_as_long(self.raw!)
+    }
+
+    func asEntid() -> Int64 {
+        defer {
+            self.raw = nil
+        }
+        return typed_value_as_entid(self.raw!)
+    }
+
+    func asKeyword() -> String {
+        defer {
+            self.raw = nil
+        }
+        return String(cString: typed_value_as_kw(self.raw!))
```

Comment:
So we have Rust copying a Rust `String` to be a C string, and then Swift copying the C string to be a Swift `String`. Please file an issue to not do that; it's a performance papercut for things like keywords, but for larger strings it'll be a potential crasher. It should be possible for us to directly turn a Rust `String` into a Swift `String`, copying no more than once.